sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
     }, \
   }
 
+/**
+ * @brief AZ_SPAN_FROM_INITIALIZED_BUFFER returns an az_span expression over an initialized byte
+ * buffer. For example
+ *
+ * uint8_t buffer[] = { 1, 2, 3 };
+ * const az_span buf = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER(buffer); // Len=3, Cap=3
+ */
 #define AZ_SPAN_FROM_INITIALIZED_BUFFER(BYTE_BUFFER) \
   (az_span) AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER(BYTE_BUFFER)
 
 /******************************  SPAN MANIPULATION */
 
 /**
- * @brief Returns a sub span of the given span.
+ * @brief az_span_slice returns a new az_span which is a sub-span of the specified span.
  *
+ * @param[in] span The original az_span
+ * @param[in] low_index An index into the original az_span indicating where the returned az_span
+ * will start
+ * @param[in] high_index An index into the original az_span indicating where the returned az_span
+ * should stop. The byte at the high_index is NOT included in the returned az_span.
+ * @return An az_span (view) into a portion (from low_index to high_index - 1) of the original
+ * az_span.
```

Comment:
We should mention what the "special" sentinel values mean like -1. Looks like a high_index of -1 means up to end (capacity and not length). We should document that and also that unlike high_index, low_index = -1 doesn't have any meaning (should it?).